vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -370,19 +383,27 @@ def collect_metrics_async(self):
                         e,
                     )
 
-    def make_batch(self, mors, metric_ids, resource_type):
+    def make_batch(
+        self,
+        mors,  # type: Iterable[vim.ManagedEntity]
+        metric_ids,  # type: List[vim.PerformanceManager.MetricId]
+        resource_type,  # type: Type[vim.ManagedEntity]
+    ):  # type: (...) -> Generator[MorBatch, None, None]
```

Comment:
We can simplify here too, as callers most likely only care that this returns an iterator (eg they don't use `gen.send()`):

```suggestion
    ):  # type: (...) -> Iterator[MorBatch]
```